workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/WorkflowDirectGraphBuilder.java

```diff
@@ -0,0 +1,229 @@
+package com.symphony.bdk.workflow.engine;
+
+import static com.symphony.bdk.workflow.engine.camunda.bpmn.builder.WorkflowNodeBpmnBuilder.DEFAULT_FORM_REPLIED_EVENT_TIMEOUT;
+
+import com.symphony.bdk.workflow.engine.WorkflowDirectGraph.Gateway;
+import com.symphony.bdk.workflow.engine.camunda.WorkflowEventToCamundaEvent;
+import com.symphony.bdk.workflow.swadl.exception.ActivityNotFoundException;
+import com.symphony.bdk.workflow.swadl.exception.InvalidActivityException;
+import com.symphony.bdk.workflow.swadl.exception.NoStartingEventException;
+import com.symphony.bdk.workflow.swadl.v1.Activity;
+import com.symphony.bdk.workflow.swadl.v1.Event;
+import com.symphony.bdk.workflow.swadl.v1.EventWithTimeout;
+import com.symphony.bdk.workflow.swadl.v1.Workflow;
+import com.symphony.bdk.workflow.swadl.v1.activity.BaseActivity;
+import com.symphony.bdk.workflow.swadl.v1.activity.RelationalEvents;
+import com.symphony.bdk.workflow.swadl.v1.event.ActivityCompletedEvent;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Builder constructs the given workflow into a direct graph structure
+ *
+ * @see WorkflowDirectGraph
+ */
+public class WorkflowDirectGraphBuilder {
+  private static final String TIMEOUT_SUFFIX = "_timeout";
+  /**
+   * element id is key, element parent id is value
+   */
+  private final Workflow workflow;
+  private final WorkflowEventToCamundaEvent eventMapper;
+
+  public WorkflowDirectGraphBuilder(Workflow workflow, WorkflowEventToCamundaEvent eventMapper) {
+    this.workflow = workflow;
+    this.eventMapper = eventMapper;
+  }
+
+  public WorkflowDirectGraph build() {
+    List<Activity> activities = workflow.getActivities();
+    WorkflowDirectGraph directGraph = new WorkflowDirectGraph();
+    activities.forEach(activity -> directGraph.registerToDictionary(activity.getActivity().getId(),
+        new WorkflowNode().id(activity.getActivity().getId())
+            .activity(activity.getActivity())
+            .elementType(WorkflowNodeType.ACTIVITY)));
+
+    for (int i = 0; i < activities.size(); i++) {
+      Activity activity = activities.get(i);
+      String activityId = activity.getActivity().getId();
+      RelationalEvents onEvents = activity.getEvents();
+      computeStandaloneActivities(activities, directGraph, i, activity, activityId, onEvents);
+      computeEvents(i, activityId, onEvents, activities, directGraph);
+    }
+    return directGraph;
+  }
+
+  private void computeStandaloneActivities(List<Activity> activities, WorkflowDirectGraph directGraph, int index,
```

Comment:
We can maybe reduce the number of parameters in this method: 
- get rid of "activityId" parameter as it can be retrieved with "activity.getActivity().getId()"
- the same for "onEvents" as it can be retrieved with "activity.getActivity().getEvents()"
- get rid of "activity" as it can be retrieved with "activities.get(index)"